File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        
        heap = []
        for cord in points:
            dist = cord[0]**2+cord[1]**2
            if len(heap)<k:
                heapq.heappush(heap,(-dist, cord))
            else:
                heapq.heappushpop(heap, (-dist, cord))
        return [x for dist, x, in heap]

File: 1014-k-closest-points-to-origin/k-closest-points-to-origin.java
import java.util.*;

class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> heap = new PriorityQueue<>((a, b) -> Integer.compare(b[0], a[0]));
        
        for (int[] cord : points) {
            int dist = cord[0] * cord[0] + cord[1] * cord[1];
            if (heap.size() < k) {
                heap.offer(new int[]{dist, cord[0], cord[1]});
            } else {
                if (dist < heap.peek()[0]) {
                    heap.poll();
                    heap.offer(new int[]{dist, cord[0], cord[1]});
                }
            }
        }
        
        int[][] result = new int[k][2];
        int idx = 0;
        while (!heap.isEmpty()) {
            int[] point = heap.poll();
            result[idx][0] = point[1];
            result[idx][1] = point[2];
            idx++;
        }
        return result;
    }
}

